sdk/src/common/env_variables.cc

```diff
@@ -0,0 +1,196 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#include "opentelemetry/sdk/common/env_variables.h"
+
+#ifdef _MSC_VER
+#  include <string.h>
+#  define strcasecmp _stricmp
+#else
+#  include <strings.h>
+#endif
+
+#include <memory>
+
+#include "opentelemetry/sdk/common/global_log_handler.h"
+#include "opentelemetry/version.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace common
+{
+
+bool GetRawEnvironmentVariable(const char *env_var_name, std::string &value)
```

Comment:
Would be good to have some unit-tests around the functions in this file. Not blocker, someone can pick this up as separate PR.

Reply:
Looking into this.

Reply:
Just in case it helps, there are some tests to validate resource detectors settings using env variables - https://github.com/open-telemetry/opentelemetry-cpp/blob/a6211fa904ac90351694616384b765c6288dfea8/sdk/test/resource/resource_test.cc#L174

Reply:
Added unit tests.